### PR TITLE
fix: [CHAOS-958]: Made hooks and components optional in ChildAppMounter

### PR DESCRIPTION
**Pull request by hrishavjha**
Signed-off-by: Hrishav <hrishav.kumar@harness.io>

### Summary

Updated `ChaosCustomMicroFrontendProps` to make `customComponents`, `customHooks` and `customFunctions` optional.

> **NOTE**: Updating `@harness/microfrontends` package to get changes from https://github.com/harness/harness-core-ui/pull/13071 and this PR(https://github.com/harness/harness-core-ui/pull/13261) which are required for development and GA release for CHAOS module.

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by SahilKr24 — DISMISSED** [review]

**Review by vanshBhatia-A4k9 — APPROVED** [review]

**Comment by SahilKr24** [comment]
retrigger coverage

**Review by knagurski — APPROVED** [review]
👍🏻

File: src/modules/75-chaos/interfaces/Chaos.types.ts [hunk 1]
```diff
@@ -10,24 +10,17 @@ import type { Diagnostic } from 'vscode-languageserver-types'
...
 export interface ChaosCustomMicroFrontendProps {
   customComponents: {
-    ConnectorReferenceField: React.ComponentType<ConnectorReferenceFieldProps>
-    OverviewChartsWithToggle: React.ComponentType<OverviewChartsWithToggleProps>
-    Duration: React.ComponentType<DurationProps>
-    NavigationCheck: React.ComponentType<NavigationCheckProps>
-  }
-  customHooks: {
-    useCreateConnectorModal: (props: UseCreateConnectorModalProps) => UseCreateConnectorModalReturn
+    ConnectorReferenceField?: React.ComponentType<ConnectorReferenceFieldProps>
```
Comment (abhinavrastogi-harness) [inline src/modules/75-chaos/interfaces/Chaos.types.ts:17]: I don't think you'll have any benefit by making existing keys optional.

It's only helpful when adding new keys.. making it optional keeps it backwards compatible, allowing independent deployments as long as you add a null check on the consumer side.

Comment (arkajyotiMukherjee) [reply src/modules/75-chaos/interfaces/Chaos.types.ts:17]: The only rationale I had was if were to remove a component later our MFE would already be forced to handle the undefined check.

**Comment by SahilKr24** [comment]
retrigger coverage

**Review by knagurski — APPROVED** [review]
👍🏻

**Review by SahilKr24 — APPROVED** [review]

File: src/modules/75-chaos/interfaces/Chaos.types.ts (hunk 1 above)
Comment (SahilKr24) [reply src/modules/75-chaos/interfaces/Chaos.types.ts:17]: @abhinavrastogi-harness shall I go ahead and merge this or remove the optional checks?

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/75-chaos/interfaces/Chaos.types.ts (hunk 1 above)
Comment (abhinavrastogi-harness) [reply src/modules/75-chaos/interfaces/Chaos.types.ts:17]: hmm okay.. lets go ahead with it then

**Pull request merged**